apps/openmw/mwmechanics/spellcasting.cpp

```diff
@@ -488,13 +489,14 @@ namespace MWMechanics
                     else if (castByPlayer)
                         MWBase::Environment::get().getWindowManager()->messageBox("#{sMagicTargetResisted}");
                 }
-
-                // If player is attempting to cast a harmful spell, show the target's HP bar
-                if (castByPlayer && target != caster)
+                else if (magicEffect->mData.mFlags & ESM::MagicEffect::Harmful && castByPlayer && target != caster)
```

Comment:
How about non-AI effects with Self range (e.g. Light or Chameleon)? These effects will be resisted too?

Reply:
Oops, I haven't thought of them.